### PR TITLE
User table

### LINKED ISSUE #16: Create users table/model
- [x] email
- [x] password_digest
- [x] role
- [x] active

### PR DESCRIPTION
**Pull request by jequeza**
### Add
* Database schema. User model, user factory and user spec.
### Update / Remove
* `Gemfile`, `rails_helper.rb`, `.gitignore` 
### Why?
* Create initial user table along with the model and respective tests. 
* Update Gemfile to include `bcrypt` gem for password security.
* Add factory bot syntax to `rails_helper.rb`
* Add coverage to `.gitignore`

Closes #16

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    it { should have_secure_password }
+  end
+
+  describe "user roles" do
+    it "user can be student user" do
+      data = {email: "sample@email.com", password: "1234", role: 0}
+      student_user = User.create(data)
+      expect(student_user).to be_valid
+      expect(student_user.role).to eq("student")
+      expect(student_user.student?).to be_truthy
```
Comment (Gvieve) [inline spec/models/user_spec.rb:20]: Very cool. I did not know be_truthy was available in capybara 🙂

**Review by Gvieve — APPROVED** [review]
Great work. Thanks so much tackling this!

File: spec/models/user_spec.rb (hunk 1 above)
Comment (jequeza) [reply spec/models/user_spec.rb:20]: capybara is a very powerful tool for sure 😄

**Comment by jequeza** [comment]
It was a fun spike! Love creating tables! BE ❤️

**Pull request merged**